model/src/main/kotlin/OrtResult.kt

```diff
@@ -403,10 +403,27 @@ data class OrtResult(
     fun getAdvisorResultsForId(id: Identifier): List<AdvisorResult> = advisorResultsById[id].orEmpty()
 
     /**
-     * Return all [RuleViolation]s contained in this [OrtResult].
+     * Return all [RuleViolation]s contained in this [OrtResult]. Optionally exclude resolved violations with
+     * [omitResolved] and remove violations below the [threshold].
      */
     @JsonIgnore
-    fun getRuleViolations(): List<RuleViolation> = evaluator?.violations.orEmpty()
+    fun getRuleViolations(omitResolved: Boolean = false, threshold: Severity = Severity.HINT): List<RuleViolation> {
```

Comment:
IMO the second argument should default to `OrtConfig.severeIssueThreshold` instead.

Reply:
That would make this a breaking change and I think that `getRuleViolation()` without any parameters should return all violations and not implicitly apply the default 'severseIssueThreshold`.

Reply:
Even if it's a breaking change, I believe that's what we need to do, to align with https://github.com/oss-review-toolkit/ort/pull/4207 and fix the general mess have have in this context.

Reply:
Why should the default value be `OrtConfig.severeIssueThreshold`? 

Above question is independent of this one: Is `OrtConfig.severeIssueThreshold` actually referring to just `OrtIssue`, no to rule violation? I'm asking this as rule violations are highly customizable (as opposed to ORT issues) and what severity actually is considered severe, in context of rule violations, is IMO up the user who implements the rules. What do you think?

Reply:
> Is `OrtConfig.severeIssueThreshold` actually referring to just `OrtIssue`, not to `RuleViolation`?

Now that you name it, I have to agree that the name "severe*Issue*Threshold" is misleading. When I introduced that configuration option, I did not intent to limit it to issues / the `OrtIssue` class, but apply it to anything that has a `Severity` (I probably was under the misbelief that `RuleViolation` inhertis from `OrtIssue`).

That's why even for the Evaluator, which creates no `OrtIssue`s at all but only `RuleViolation`s, we currently have this code:

https://github.com/oss-review-toolkit/ort/blob/42845fa5eb5fbb1a9bc617aa778c2979ca0da26d/cli/src/main/kotlin/commands/EvaluatorCommand.kt#L304-L305

However, we also had already internal discussion about the multiple ways now to tweak the effect of a severity for rule violations: Either by changing the emitted `Severity` in the `rules.kts` directly, or by altering the `severeIssueThreshold`, and how this all should refer to the exit code of the Evaluator.

I guess we need some more brainstorming here.

Reply:
> However, we also had already internal discussion about the multiple ways now to tweak the effect of a severity for rule violations: Either by changing the emitted `Severity` in the `rules.kts` directly, or by altering the `severeIssueThreshold`, and how this all should refer to the exit code of the Evaluator.

While I agree that this is not ideal, I don't think this is strictly relevant for this PR.

After all I still think that the default behaviour of `getRuleViolations()` should return all violations, and not adhere to the configured `severeIssueThreshold`.

Reply:
Hmm, I guess I'd be fine with not having a default value at all for the `severity` parameter (just like in the `concludeSeverityStats()` function), but I'd not like to have a default value now and eventually change that default value later (as that makes it hard for callers who rely on the default to check whether they still work as intended).

Reply:
How about making `minSeverity` nullable and defaulting to `null`?
Which would then mean that no filter at all is applied?

Reply:
> After all I still think that the default behaviour of `getRuleViolations()` should return all violations

That makes sense. However, the current implementation assumes that `Severity.HINT` is the lowest severity, and that filtering with `it.severity >= minSeverity` actually keeps all violations.

How about instead making `minSeverity` a nullable parameter and not filter at all if it's `null`?

Reply:
> How about instead making `minSeverity` a nullable parameter and not filter at all if it's `null`?

I think in this case it's also ok to use `Severity.values()[0] `, just as alternative idea to avoid nullability.

Reply:
But that would still filter potentially unnecessarily, and it makes the assumption that the least severe enum value is the first entry (the values could also be sorted exactly the opposite).

Reply:
> But that would still filter potentially unnecessarily

I don't think performance tweaks are needed here

> , and it makes the assumption that the least severe enum value is the first entry (the values could also be sorted exactly the opposite).

Assumption about the ordering is made `as-is` already, and IMO it's ok to make assumptions about the ordering.

Reply:
> I don't think performance tweaks are needed here

I agree they're not needed. But I believe it's still good practice to write code so that unnecessary work is avoided (if readability is not sacrificed).